src/libs/DebugUtils.ts

```diff
@@ -0,0 +1,587 @@
+    if (data === undefined) {
+        return text === 'undefined';
+    }
+
+    if (typeof data === 'object') {
+        return text === stringifyJSON(data as Record<string, unknown>);
+    }
+
+    return text === String(data);
+}
+
+/**
+ * Determines the number of lines needed to display the data.
+ *
+ * @param data - string representation
+ * @returns number of lines needed to display the data
+ */
+function getNumberOfLinesFromString(data: string) {
+    return data.split('\n').length || 1;
+}
+
+/**
+ * Converts every value from an onyx data object into it's string representation, to be used as draft data.
+ *
+ * @param data - onyx data object
+ * @returns converted data object
+ */
+function onyxDataToDraftData(data: OnyxEntry<Record<string, unknown>>) {
+    return Object.fromEntries(Object.entries(data ?? {}).map(([key, value]) => [key, onyxDataToString(value)]));
+}
+
+/**
+ * Validates if a string is a valid representation of a number.
+ */
+function validateNumber(value: string) {
+    if (value === 'undefined' || value === '' || (!value.includes(' ') && !Number.isNaN(Number(value)))) {
+        return;
+    }
+
+    throw new NumberError();
+}
+
+/**
+ * Validates if a string is a valid representation of a boolean.
+ */
+function validateBoolean(value: string) {
+    if (OPTIONAL_BOOLEAN_STRINGS.includes(value)) {
+        return;
+    }
+
+    throw new SyntaxError('debug.invalidValue', {cause: {expectedValues: OPTIONAL_BOOLEAN_STRINGS.join(' | ')}});
+}
+
+/**
+ * Validates if a string is a valid representation of a date.
+ */
+function validateDate(value: string) {
+    if (value === 'undefined' || (isMatch(value, CONST.DATE.FNS_DB_FORMAT_STRING) && isValid(new Date(value)))) {
+        return;
+    }
+
+    throw new SyntaxError('debug.invalidValue', {cause: {expectedValues: CONST.DATE.FNS_DB_FORMAT_STRING}});
+}
+
+/**
+ * Validates if a string is a valid representation of an enum value.
+ */
+function validateConstantEnum(value: string, constEnum: ConstantEnum) {
+    const enumValues = Object.values(constEnum).flatMap((val) => {
+        if (val && typeof val === 'object') {
+            return Object.values(val).map(String);
+        }
+        return String(val);
+    });
+
+    if (value === 'undefined' || value === '' || enumValues.includes(value)) {
+        return;
+    }
+
+    throw new SyntaxError('debug.invalidValue', {cause: {expectedValues: `${enumValues.join(' | ')} | undefined`}});
+}
+
+/**
+ * Validates if a string is a valid representation of an array.
+ */
+function validateArray(
+    value: string,
+    arrayType:
+        | 'string'
+        | 'number'
+        | 'boolean'
+        // Constant enum
+        | ConstantEnum
+        // Object
+        | Record<
+              string,
+              | 'string'
+              | 'number'
+              | 'object'
+              | 'boolean'
+              | 'array'
+              | PropertyTypes
+              // Constant enum
```

Comment:
Why are these comments needed?

Reply:
They're not, I'm removing them thanks! 😄